src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -446,7 +447,9 @@ public function toggleStatusAction(Request $request, $categoryId)
         }
 
         try {
-            $command = new ToggleCategoryStatusCommand((int) $categoryId);
+            $isEnabled = $this->getQueryBus()->handle(new GetCategoryStatus((int) $categoryId));
+
+            $command = new ToggleCategoryStatusCommand((int) $categoryId, !$isEnabled);
```

Comment:
@matks I've updated `ToggleCategoryStatusCommand` to accept new status and behave in predictable way. 

However, now I think that `Toggle` is not the best word here. It actually does not toggle anything, but instead updates/set new status. So I think it should be named `SetCategoryStatusCommand` or similar instead, wdyt?

Secondly, see line above `$isEnabled = $this->getQueryBus()->handle(...);`, I don't know whether it's a problem or not, but I don't think we should get status using Query. This basically gets single attribute of `Category`. What if I need both status and name, should I create `GetStatusAndName` Query? I think these generic data retrieval should be improved somehow.

What if we use read model repositories instead. It would look something like this:

```php
namespace Domain\Category\Repository\ReadModel;
// or namespace Domain\Category\ReadModel\Repository;

interface CategoryRepositoryInterface
{
    public function getNameById($categoryId): string;

    public function getStatusById($categoryId): bool;
}
```
And the implementation could be done either using adapters or plain SQL, like:

```php
namespace Domain\Category\Repository\ReadModel;

final class CategoryRepository implements CategoryRepositoryInterface
{
    // ...

    public function getNameById($categoryId)
    {
        $sql = 'SELECT name from category ......';
      
        return $this->connection->select($sql);
    }
}
```

Read model repositories **would not be meant to deal with domain entities** (which we don't have yet), but rather with reading data and could be injected into query handlers if we need to compose more complex (for specific use case) data read using multiple read model repositories.

Also, this would allow to implement read model repositories without depending on legacy classes, as we could write plain SQLs already.

In other domains, let's say `Order` we may have multiple read model repositories (e.g. `OrderRepository`, `OrderPaymentRepository`, `OrderInvoiceRepository` and etc).

I think this approach could (you can prove me being wrong :smile:) help avoid creating many generic queries for reading data and instead encapsulate generic data retrieval into read model repositories.

Reply:
> However, now I think that Toggle is not the best word here. It actually does not toggle anything, but instead updates/set new status. So I think it should be named SetCategoryStatusCommand or similar instead, wdyt?
That's a yes 👍let's get rid of Toggle